docs/cluster-agent/HORIZONTAL_POD_AUTOSCALING.md

```diff
@@ -0,0 +1,292 @@
+`kubectl apply -f manifests/cluster-agent/hpa-example/cluster-agent-hpa-svc.yaml`
+
+Note that the first service is used for the communication between the node agents and the DCA but the second is used by Kubernetes to register the External Metrics Provider.
+
+At this point you should be having:
+
+```
+PODS:
+
+NAMESPACE     NAME                                     READY     STATUS    RESTARTS   AGE
+default       datadog-cluster-agent-7b7f6d5547-cmdtc   1/1       Running   0          28m
+
+SVCS:
+
+NAMESPACE     NAME                  TYPE        CLUSTER-IP        EXTERNAL-IP   PORT(S)         AGE
+default       datadog-cluster-hpa   ClusterIP   192.168.254.87    <none>        443/TCP         28m
+default       dca                   ClusterIP   192.168.254.197   <none>        5005/TCP        28m
+
+```
+
+### Register the External Metrics Provider
+
+Once the DCA is up and running, you can register it as an External Metrics Provider, via the service exposing the port 443.
+
+To do so, simply apply the following RBAC rules:
+
+`kubectl apply -f manifest/hpa-example/rbac-hpa.yaml`
+
+```
+clusterrolebinding.rbac.authorization.k8s.io "system:auth-delegator" created
+rolebinding.rbac.authorization.k8s.io "dca" created
+apiservice.apiregistration.k8s.io "v1beta1.external.metrics.k8s.io" created
+clusterrole.rbac.authorization.k8s.io "external-metrics-reader" created
+clusterrolebinding.rbac.authorization.k8s.io "external-metrics-reader" created
+```
+ 
+Once you have the DCA running and the service registered, you can create an HPA manifest and let the DCA pull metrics from Datadog.
+The following part of the walk through explains how you can set up your agents in order to collect metrics from the applications running on your cluster.
+Those metrics will then be available for you to autoscale, via the DCA, the resources of your cluster.
+If your agents are already instrumented and configured to communicate with the DCA, you can directly jump to the [running the hpa]() section.
+
+## Spinning up your agents
+
+The first step to spin up your node agents is to create the RBAC rules giving them access to the API of the local kubelet.
+
+`kubectl apply -f manifests/cluster-agent/rbac-agent.yaml`
+
+This will create a Service Account, a Cluster Role and a Cluster Role Binding.
+
+```
+clusterrole.rbac.authorization.k8s.io "datadog-agent" created
+serviceaccount "datadog-agent" created
+clusterrolebinding.rbac.authorization.k8s.io "datadog-agent" created
+```
+
+Then, you will just need to update the manifest of the DaemonSet with your API_KEY.  Make sure the `DD_CLUSTER_AGENT` option is set to true and the secret you referenced in the manifest of the DCA to enable the communication between the node agent and the DCA is also set.
+Once this is configured, you can create the resources:
+
+`kubectl apply manifest/agent.yaml`
+
+```
+daemonset.extensions "datadog-agent" created
+```
```

Comment:
I think this section (Spinning up your agents) also may not be belong in this doc since this is general information about how to run the node agents. I think the assumption would be that you already have datadog up and running before looking into autoscaling?

What do you think?

Reply:
I agree in the end - I'm refactoring